### ISSUE TITLE
"User In Earmuff Mode" ProtoFlux node

### ISSUE THREAD

**Issue by paradoxical-autumn** [Yellow-Dog-Man/Resonite-Issues#1913]
### Is your feature request related to a problem? Please describe.

Currently, you cannot get if a specific user is in earmuff mode easily and I believe that a protoflux node to get this state would be helpful (such as on avatars)

### Describe the solution you'd like

A ProtoFlux node which takes a user as input and outputs a boolean for if they are currently in earmuff mode

### Describe alternatives you've considered

Using a `SettingValueSync<bool, EamuffSettings>`, a cloud variable and a facet to get this information outside of userspace. This wouldn't allow for arbitrary access to the setting though.

### Additional Context

Ideally, the node should be something like this:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/110392755/955067a7-df9f-44d8-9246-2e55e478fde4)

### Requesters

paradoxical_autumn (Discord) // paradox19 (IGN)

**Comment by Frooxius** [comment]
What's the use-cases for this? Why do you want to know if the user is in an earmuff mode?

I'm not sure if this is something that's necessarily good to expose, since it might be something the user wants to keep private(ish).

**Comment by paradoxical-autumn** [comment]
I was mainly thinking about avatar customisations (such as toggling an effect / mesh) and extra world functionality (such as informing people if a user is on earmuff mode) but I understand your concerns w/ user privacy.

I was mainly looking at VRChat's implementation of earmuff mode and how it's _always_ displayed to everyone on a user's nameplate.

**Comment by JackTheFoxOtter** [comment]
I somewhat agree with the privacy argument, but then again, I don't think exposing this would be very harmful. Whisper voice mode is also exposed in a similar way.

**Comment by Frooxius** [comment]
We could do that yeah. 

I suppose we could just add a privacy toggle for this too. So you can toggle if the world can know if you're in ear-muff or not, that might give best from both world?

**Comment by paradoxical-autumn** [comment]
That'd be fine by me!

**Comment by Frooxius** [comment]
Ok, we'll tackle it that way! Thank you!

**Comment by JackTheFoxOtter** [comment]
> I suppose we could just add a privacy toggle for this too. So you can toggle if the world can know if you're in ear-muff or not, that might give best from both world?

Perhaps, even if it sounds a little excessive. The main point of having an indicator for the whisper bubble is so other people know it's active and can adjust accordingly to the person potentially not hearing people outside of it. I'm not quite sure how much earmuff mode deviates from that.

**Comment by paradoxical-autumn** [comment]
> It sounds excessive
While I agree with this statement @JackTheFoxOtter, having an option to turn off the feature is better than no option at all for people who might not want this shown (such as if they _need_ the feature (like me) but don't want it publicly known).

I'm fine with it as long as you're opted in by default (like how when the "opt out of media metadata" feature was released, everyone had it defaulted to opt in)

**Comment by paradoxical-autumn** [comment]
> It sounds excessive
While I agree with this statement @JackTheFoxOtter, having an option to turn off the feature is better than no option at all for people who might not want this shown (such as if they _need_ the feature (like me) but don't want it publicly known).

I'm fine with it as long as you're opted in by default (like how when the "opt out of media metadata" feature was released, everyone had it defaulted to opt in)

And also with the whisper bubble, you _can_ turn that off by editing your avatar so this isn't much different

**Comment by JackTheFoxOtter** [comment]
I guess it might be more similar to client volume settings than voice modes, yeah. I didn't consider people leaving it on constantly for accessibility reasons.

**Comment by JackTheFoxOtter** [comment]
> And also with the whisper bubble, you _can_ turn that off by editing your avatar so this isn't much different

Well, no, because you can't change what your voice mode reports as for the ProtoFlux nodes / user inspector.

**Comment by paradoxical-autumn** [comment]
> Well, no, because you can't change what your voice mode reports as for the ProtoFlux nodes / user inspector.

That's true, I completely forgot about how that node isn't just "Is muted"

**Comment by Frooxius** [comment]
The difference here is that whisper bubble is something that's part of the "in-world" voice communication mechanism. It's temporary and doesn't transfer from world to world or persist.

Earmuff mode is global and persistent accessibility setting, so I'm not too sure if it's something users necessarily want to have exposed about themselves. 

My thinking about it is more along terms of accessibility settings. Say when we introduce "Photosensitive" setting, which will disable certain effects - but do it in a way that's not detectable. That might also be something that users might not want to have exposed about themselves.

**Comment by shiftyscales** [comment]
Personally I don't think I'd want to see this state exposed at all. Fundamentally- it's meant to be an accessibility setting that a user tunes to their own personal comfort.

Other users have no need to know who is using earmuff mode, nor how it is configured. It is a personal preference like with the volume levels which are also not exposed in any capacity.

If a user wishes to globalize this value, there are other existing mechanisms they could use to do so, but I don't think this should be something we should support as an official feature, even if we include an opt-in setting.

**Comment by Frooxius** [comment]
With the proposed solution, if you wish to not expose this value for yourself, you'll have that option.

There are always ways to expose things manually, but it requires additional effort and complexity from the user. If this is something that's common enough, it might be worth adding (e.g. similarly how you can query the voice status).

**Comment by shiftyscales** [comment]
Per your above example though around whisper bubbles- voice status is something that applies directly to worldspace, as worldspace is responsible for controlling/limiting the available voice modes based on the world's configuration, @Frooxius. As such they aren't directly comparable.

At the least, if it were added, it should be opt-in (disabled by default) rather than opt-out. Leave it to the user to choose to share if they are making use of accessibility features rather than publishing that info publicly by default.

**Comment by paradoxical-autumn** [comment]
Shifty raises a good point and I'm gonna say that I actually believe that it should be opt-in, so no one is randomly thrust into a state they're not comfortable with.

**Comment by Frooxius** [comment]
@shiftyscales Yes, there's differences, but I think that's kind of missing the point.

There is clearly interest in using this to display on avatar and such in this - same way people want to display muted status or dash. Therefore in that regard it *is* comparable. 

I understand that you are not interested in sharing this with the world, but @paradoxical-autumn clearly is.

**Comment by paradoxical-autumn** [comment]
Another reason why I think this is a helpful feature is as I've had multiple people think I'm ignoring them when in actuality I can't hear them due to having the mode enabled

**Comment by Azavit** [comment]
Opt In: means the value is nearly useless if you aren't the one wanting it to be set up and will cause confusion when a person tries to get it working and it just fails as you may not be focused on the settings menu as much when building with Protoflux.  Also means 90% of people won't check the option (for every button you need to press you loose 90% of the people) even if they don't care about others knowing about it.

Opt Out: The person will already be in the settings menu and will already be thinking "hey I don't want this transmitted." and opts out intentionally and will most likely remember that "yes I opted out, that's why this node doesn't work" vs others who may not have even noticed the setting to opt in.

**Comment by paradoxical-autumn** [comment]
I feel like this issue is going to go round in circles of "should it be opt-out or opt-in?" as there are valid arguments for both sides.

**_Opt-out pros:_**
- It'll automatically work for everyone
- People can turn it off
**_Opt-out cons:_**
- People might not realise that this feature released and be thrust into a state of discomfort 

**_Opt-in pros:_**
- Users don't have to worry about this change as their privacy is respected
- Anyone who doesn't understand what this setting does can just leave it on default without worrying about their privacy
**_Opt-in cons:_**
- 90% of people won't know about this existing and be confused as to how it works for <10% of people
- The node is basically useless for arbitrary access as it won't work 90% of the time

# Conclusion
This feature request will need more feedback from the community to find out **how many people want it to be shown** and **how many people want it to be hidden**

If you've made it this far, all I can ask is that you please leave your feedback on this. You don't need to write an essay, just a simple "I want this to be opt-out" or "I want this to be opt-in". Thank you!

**Comment by Frooxius** [comment]
I think it depends a bit on use cases. From the sound of it, most of the time people will want to setup their avatar with some indicator to show this, rather than other users detecting if someone is in earmuff.

If those are the main cases, then the burden to opt-in should be on the user setting up their avatar - it's up to them to read up on the documentation on the node and see how it works, since they're the ones putting effort to setup something.

**Comment by quellus** [comment]
A bit late to the party, but my use case is also an indicator. I'd like to be able to indicate to others to avoid people thinking I'm ignoring them when I just can't hear them.

**Comment by shiftyscales** [comment]
Having used the earmuff feature considerable more since my initial impressions- I rescind my prior remarks. I think it would be helpful as a default to display earmuff state in some form so that users can be made aware that if they wish to speak to you, they'd be best to do so from the front / in close range.

It could also be a useful addition for the default platform name plate. #3820.

**Comment by Draconas** [comment]
updooting this. I have indicators on my avatars that im in my dash, lagging, not in my headset, even battery levels, im surprised there isnt an easy way to read or write the users earmuff state beyond the toggle in settings, being able to tell nearby people that they have to be in range to be heard would very helpful.

**Comment by Redd56** [comment]
Had a thought. What if nullable boolean. The "value?" Part aka null or not part being for if they have privacy mode on or off and the actual bool value for if they have earmuff mode on

**Comment by fulgenscode** [comment]
Given how many times I've seen people silently run away from someone talking because of earmuff mode, I don't think Resonite should assume users want to hide this, essentially assuming that users prefer to opt out of good communication.

I think it should default to on for newly created accounts and if it's a problem for a lot of current users it should default to off for accounts that already exist.

**Comment by TisFoolish** [comment]
Defaulting to on would be confusing for new users and differ from other platforms.

**Comment by Redd56** [comment]
How would it be different from other platforms or confusing. Vrchat doesn't even to my knowledge have a way to turn off the icon in the nameplate

**Comment by TisFoolish** [comment]
I mistook them saying earmuff mode should default to on, my bad

**Comment by Redd56** [comment]
ah, your good just was makin sure we were all on the same page
